txn_writer.go

```diff
@@ -288,17 +284,12 @@ func (payload SafeWriterPayload) validate(root string, sm gps.SourceManager) err
 // This mostly guarantees that dep cannot exit with a partial write that would
 // leave an undefined state on disk.
 func (sw *SafeWriter) Write(root string, sm gps.SourceManager, noExamples bool) error {
-
-	if sw.Payload == nil {
-		return errors.New("Cannot call SafeWriter.Write before SafeWriter.Prepare")
-	}
```

Comment:
@sdboyer removing this section is the only bit I'm unsure on ... how do we ensure that someone has used `NewSafeWriter` rather than just created a `SafeWriter`?